sound/soc/sof/sof-audio.c

```diff
@@ -669,28 +669,17 @@ int sof_set_up_pipelines(struct device *dev, bool verify)
 }
 
 /*
- * This function doesn't free widgets during suspend. It only resets the set up status for all
- * routes and use_count for all widgets.
+ * Helper to tear down the pipelines. This should be called to free widgets
+ * when needed, e.g. at suspend, or critical error hit during initialization.
```

Comment:
I really don't see why you removed the 'verify' case, this is useful for the first boot and first suspend so that you can fully check that the topology is correct.
We added this for a purpose. Please don't come and nuke what others worked on.

Reply:
> I really don't see why you removed the 'verify' case, this is useful for the first boot and first suspend so that you can fully check that the topology is correct.
> We added this for a purpose. Please don't come and nuke what others worked on.

Oh, I actually don't want to change any line of it in case multi-core works fine with it, :). 

I realize this PR looks invasive and that's why it is tagged with 'RFC'. This piece of code looks convoluted to me and I was compulsive to make it simple. @ranj063 I need your help on the PR, let's sync offline.